clang/lib/Serialization/ASTWriter.cpp

```diff
@@ -1115,13 +1115,13 @@ void ASTWriter::WriteBlockInfoBlock() {
 }
 
 /// Prepares a path for being written to an AST file by converting it
-/// to an absolute path and removing nested './'s.
+/// to an absolute path and removing nested './'s and '../'s.
 ///
 /// \return \c true if the path was changed.
 static bool cleanPathForOutput(FileManager &FileMgr,
                                SmallVectorImpl<char> &Path) {
   bool Changed = FileMgr.makeAbsolutePath(Path);
-  return Changed | llvm::sys::path::remove_dots(Path);
+  return Changed | llvm::sys::path::remove_dots(Path, true);
```

Comment:
Removing `..` is only correct if you know the containing directory cannot be a symlink.  Do we actually know that here?

Reply:
Hm I am not sure actually!

Reply:
The redirecting filesystem itself removes .. (which is maybe not 100% correct either but it's more limited there than it would be here I think). Can you clarify why that's not sufficient? I think I must be missing why this is needed.

Reply:
You are right. We must have done this before using the VFS and not revisited. Lemme undo this .. addition!

Reply:
Removed!